### PR TITLE
[Doc] Make the launching ray clusters doc page title more reflective

### PR DESCRIPTION
**Pull request by jjyao**
Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>

## Why are these changes needed?
Make the title more clear that it's about launching clusters in the cloud or on-prem.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/vms/user-guides/launching-clusters/index.rst [hunk 1]
```diff
@@ -1,9 +1,9 @@
 .. _launching-vm-clusters:
 
-Launching Ray Clusters
-======================
+Launching Ray Clusters in the Cloud (AWS, GCP, Azure) or On-Premise
```
Comment (DmitriGekhtman) [inline doc/source/cluster/vms/user-guides/launching-clusters/index.rst:3]: Nits: 

Should be either On-Premises for the full word or On-Prem for the abbreviation.

Does the title appear too long in the doc sidebar, or is it ok? I haven't taken a look at the rendered docs.
If it needs to be shortened, "Launching" is redundant and "Ray" might be redundant -- one or both words could potential be removed.
On-Prem in place of the full word On-Premises would also keep it shorter.

Comment (richardliaw) [reply doc/source/cluster/vms/user-guides/launching-clusters/index.rst:3]: How about:

Ray Clusters on AWS / GCP / Azure / On-Prem

Comment (jjyao) [reply doc/source/cluster/vms/user-guides/launching-clusters/index.rst]: sidebar looks ok. I changed to `Launching Ray Clusters on AWS, GCP, Azure, On-Prem` to be consistent with sub-page titles.

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
thank you!

**Pull request merged**